homeassistant/components/tami4/button.py

```diff
@@ -24,24 +24,36 @@ class Tami4EdgeButtonEntityDescription(ButtonEntityDescription):
     press_fn: Callable[[Tami4EdgeAPI], None]
 
 
-BUTTONS: tuple[Tami4EdgeButtonEntityDescription] = (
-    Tami4EdgeButtonEntityDescription(
-        key="boil_water",
-        translation_key="boil_water",
-        press_fn=lambda api: api.boil_water(),
-    ),
+BOIL_WATER_BUTTON = Tami4EdgeButtonEntityDescription(
+    key="boil_water",
+    translation_key="boil_water",
+    press_fn=lambda api: api.boil_water(),
 )
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Perform the setup for Tami4Edge."""
-    api: Tami4EdgeAPI = hass.data[DOMAIN][entry.entry_id][API]
 
-    async_add_entities(
-        Tami4EdgeButton(api, entity_description) for entity_description in BUTTONS
-    )
+    api: Tami4EdgeAPI = hass.data[DOMAIN][entry.entry_id][API]
+    device = await hass.async_add_executor_job(api.get_device)
+    drinks = device.drinks
+
+    buttons = [
+        Tami4EdgeButton(
+            api=api,
+            entity_description=Tami4EdgeButtonEntityDescription(
+                key=drink.id,
+                name=drink.name,
+                press_fn=lambda api, drink=drink: api.prepare_drink(drink),  # type: ignore[misc]
```

Comment:
```suggestion
                press_fn=lambda api: api.prepare_drink(drink),
```

Reply:
This change produces an error `B023 Function definition does not bind loop variable drink` because this can cause late binding

Reply:
In that case you need to disregard my previous comments and we do need a second extension, sorry for that, didn't know this wasn't possible

Reply:
I see that `type: ignored` is used a lot in the codebase. I'm thinking leaving it as is